File: frontend/src/api/reviewService.js
import api from "./authInterceptor";
import publicApi from "./publicApi";

const reviewService = {
  // ✅ PUBLIC
  getHotelReviews: (hotelId) => 
    publicApi.get(`/api/reviews/hotel/${hotelId}`),

  getHotelRatingSummary: (hotelId) =>
    publicApi.get(`/api/reviews/hotel/${hotelId}/rating-summary`),

  getAllReviews: () =>
    publicApi.get(`/api/reviews`),

  // 🔒 AUTH REQUIRED
  canWriteReview: (hotelId) => 
    api.get(`/api/bookings/can-review/${hotelId}`),

  getReviewableHotels: () =>
    api.get("/api/bookings/can-review"),

  createReview: (payload) => 
    api.post("/api/reviews", payload),

  updateReview: (id, payload) => 
    api.put(`/api/reviews/${id}`, payload),

  deleteReview: (id) => 
    api.delete(`/api/reviews/${id}`),

  getReviewsByUser: (id) =>
    api.get(`/api/reviews/user/${id}`),

  hasCompletedBooking: () =>
    api.get(`/api/bookings/has-completed-booking`)
};

export default reviewService;

File: backend/src/main/java/com/example/appdev/wildfire/service/ReviewService.java
package com.example.appdev.wildfire.service;

import com.example.appdev.wildfire.dto.hotel.HotelRatingSummaryDTO;
import com.example.appdev.wildfire.dto.review.ReviewRequestDTO;
import com.example.appdev.wildfire.entity.Review;
import com.example.appdev.wildfire.entity.User;

import java.util.List;

public interface ReviewService {

    Review createReview(ReviewRequestDTO dto, User user);

    List<Review> getReviewsByHotel(Long hotelId);

    List<Review> getReviewsByUser(Long userId);

    void deleteReview(Long id);

    void deleteOwnedReview(Long reviewId, User user);

    Double getAverageRatingByHotel(Long hotelId);

    HotelRatingSummaryDTO getHotelRatingSummary(Long hotelId);

    Review updateReview(Long reviewId, ReviewRequestDTO dto, User user);

    List<Review> getAllReviews();

    long countReviewsByUser(Long userId);
}
